projects/packages/installer/src/class-manager.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * The installer manager class file.
+ *
+ * @package automattic/jetpack-installer
+ */
+
+namespace Automattic\Jetpack\Installer;
+
+use Composer\Installer\LibraryInstaller;
+use Composer\Package\PackageInterface;
+
+/**
+ * The manager class that is the main agent for the installer plugin.
+ * It handles the installation process by accepting the installed package
+ * and modifying its installation path.
+ * */
+class Manager extends LibraryInstaller {
+
+	/**
+	 * Receives a supported package instance and modifies its installation path.
+	 *
+	 * @param PackageInterface $package the installed package object.
+	 * @throws \InvalidArgumentException $exception When the package is not the expected type.
+	 * @return String modified installation path.
+	 */
+	public function getInstallPath( PackageInterface $package ) {
+		$prefix = substr( $package->getPrettyName(), 0, 19 );
+		if ( 'automattic/jetpack-' !== $prefix ) {
+			throw new \InvalidArgumentException(
+				'Unable to install package: the Jetpack installer '
+				. 'plugin only supports packages from the Jetpack monorepo '
+				. '"automattic/jetpack-"'
+			);
+		}
```

Comment:
I see the example at https://getcomposer.org/doc/articles/custom-installers.md has something like this, but that's because it strips the prefix. We're not stripping a prefix, and there's not really a reason for us to do so.

Reply:
Yeah, you have a point - the `supports` method defines what packages go through this.